client/src/Movies/Imdb.js

```diff
@@ -0,0 +1,48 @@
+import React from "react";
+//imports a random movie or TV show from IMDB api...
+class Imdb extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      movieList: []
+    };
+  }
+  componentDidMount = () => {
+    fetch(
+      "https://www.omdbapi.com/?apikey=8894cd6c&plot=full&i=tt0" +
+        `${Math.floor(Math.random() * (999999 - 100000) + 100000)}`
+    )
+      .then(res => res.json())
```

Comment:
Ok, so I guess error handling is tricky with fetch. It doesn't catch an error unless it is a network error, which doesn't handle our 400/500 status codes. So, in this .then we need to 
```suggestion
      .then(res => {
        if (res.ok) {
          return res.json();
        }
        else {
          throw new Error("Couldn't get data from api");
        }
      })
```
https://stackoverflow.com/questions/38235715/fetch-reject-promise-and-catch-the-error-if-status-is-not-ok
Ok, but throw new Error just like, stops everything. Which is ok, but not very user friendly. If we just console.error the error or something instead, the code continues to run to the next then block, and we set the state with the response object, even though all it contains is failure information. So your this.state.movieList ends up being an array with a useless object on it, which makes your condition checking in render difficult. I would recommend adding an error state, so instead of throwing a new error, you set the error state to true. Then you can use it to see if you should set the state in the next then block. Then you can even use your error state to render a custom error message if the fetch fails.